dist/route-recognizer.js

```diff
@@ -153,7 +153,7 @@
     function $$route$recognizer$$DynamicSegment(name) { this.name = name; }
     $$route$recognizer$$DynamicSegment.prototype = {
       eachChar: function(currentState) {
-        return currentState.put({ invalidChars: "/", repeat: true });
+        return currentState.put({ validChars: undefined, invalidChars: "/", repeat: true });
```

Comment:
@chadhietala this should likely also be added to the perf guide. Order of key insertion can matter etc.